C3real-kill3r/H-digest
# #167288641 Enable use to verify their account

**What does this PR do?**

- [x] refactored registration view and serializer
- [x] created a custom user model
- [x] moved the templates and statics folder to root directory
- [x] refactored login serializer and view
- [x] add account verification view

**Any background context you want to provide or addition**

**How to test**

- clone this repo `git clone https://github.com/C3real-kill3r/NCPWD.git`
- checkout to this branch `git checkout ft-account-confirmation-167288641`
- run the server `python manage.py runserver`
- register a user `http://127.0.0.1:8000/api/users/`
```
{"user":{
	"first_name":"John",
	"last_name":"Doe",
	"username":"Jdoe",
	"email":"your_email@email.com",
	"password":"Try1234@"
}}
```
- verify your account `http://127.0.0.1:8000/api/account/verify/generated_token/uid/`

- login the user `http://127.0.0.1:8000/api/users/login/`
```
{"user":{
	"email":"your_email@email.com",
	"password":"Try1234@"
}}
```

**What are the relevant pivotal tracker stories?**

[#167288641](https://www.pivotaltracker.com/story/show/167288641)

**Note:**

I had to edit the authentication app for the activation feature to work. @caveinn
